fe/projects/Seller.Portal/src/shared/components/Catalog/Catalog.js

```diff
@@ -177,27 +178,201 @@ function Catalog(props) {
+
+        return fetch(url, requestOptions)
+            .then(function (response) {
+                dispatch({ type: "SET_IS_LOADING", payload: false })
+
+                return response.json().then((jsonResponse) => {
+
+                    if (!response.ok) {
+                        toast.error(jsonResponse.message);
+                    }
+                });
+            })
+    };
+
     const copyToClipboard = (text) => {
         ClipboardHelper.copyToClipboard(text);
-    }
+    };
 
     const handleQRCodeDialog = (item) => {
         setSelectedItem(item);
         setOpenQRCodeDialog(true);
+    };
+
+    const onDragStart = (result) => {
+        const item = items.find((x) => x.id === result.draggableId);
+        item ? setDraggingItem(item) : setDraggingItem({});
+
+        const queryAttr = "data-rbd-drag-handle-draggable-id";
+        const domQuery = `[${queryAttr}='${result.draggableId}']`;
+        const draggedDOM = document.querySelector(domQuery);
+
+        setPlaceholderProps({ position: result.source.index, height: draggedDOM.clientHeight + 5});
+    };
+
+    const onDragEnd = (result) => {
+        setIsDragableDisable(true);
+        setPlaceholderProps({});
+
+        const { destination, source, draggableId } = result;
+
+        if (!destination) {
+            setDraggingItem({});
+            return;
+        }
+
+        if (
+            destination.droppableId === source.droppableId &&
+            destination.index === source.index
+        ) {
+            setDraggingItem({});
+            return;
+        }
+
+        const newCategoryArray = reorder(items, source.index, destination.index);
+        handleChangeEntityOrder({ id: draggableId, order: (destination.index + 1) + (page * props.defaultItemsPerPage) });
+        setItems(newCategoryArray);
+
+        setDraggingItem({});
+    };
+
+    const reorder = (list, source, destination) => {
+        const result = Array.from(list);
+        result.splice(source, 1);
+        result.splice(destination, 0, draggingItem);
+
+        return result;
+    };
+
+    const handleChangedPageWhenDragItem = (toPage) => {
+        if (draggingItem && !isDragableDisable) {
+            handleChangePage(null, toPage);
+        }
+    };
+
+    const disablePrevChangePageArea = () => {
+        return page == 0 ? true : false;
+    };
+
+    const disableNextChangePageArea = () => {
+        return page == props.pagedItems.pageCount - 1 ? true : false;
+    };
+
+    const tableRow = (provided, item) => {
+        return (
+            <TableRow
+                ref={provided.innerRef}
+                {...provided.draggableProps}
+                {...provided.dragHandleProps}
+                className="catalog__table__row"
+            >
+                {props.table.actions &&
+                    <TableCell width="12%" height={placeholderProps.height}>
+                        {props.isDragDropEnable && (window != undefined && window.innerWidth > 800) &&
```

Comment:
**suggestion** I would remove isDragDropEnable completely and use the actions property instead - line 295